packages/babel-generator/src/printer.ts

```diff
@@ -465,6 +465,16 @@ class Printer {
     return this._indentRepeat * this._indent;
   }
 
+  ensureNoLineTerminator(fn: () => void) {
+    const { _noLineTerminator } = this;
+    this._noLineTerminator = true;
+    try {
+      fn();
+    } finally {
+      this._noLineTerminator = _noLineTerminator;
+    }
+  }
```

Comment:
It seems that currently our generator is not allowed to throw exceptions, I personally prefer not to have this try.